lib/email/styles.rb

```diff
@@ -237,10 +236,40 @@ def plugin_styles
       @@plugin_callbacks.each { |block| block.call(@fragment, @opts) }
     end
 
+    def inline_secure_images(attachments)
```

Comment:
I feel this method is a little ugly with lots of data munging, suggestions to clean up would be welcome.